### ISSUE TITLE
[BUG] [iopaint] with local_files_only parm, sd model failed to load

### ISSUE THREAD

**Issue by huangzhike** [Sanster/lama-cleaner#423]
**Model**
"Sanster/PowerPaint-V1-stable-diffusion-inpainting"
"kandinsky-community/kandinsky-2-2-decoder-inpaint"
"diffusers/stable-diffusion-xl-1.0-inpainting-0.1"

**Describe the bug**

starting the iopaint app with ` local_files_only=True`, and the parm will not be passed to huggingface model, which causes the model fails to load.

![image](https://github.com/Sanster/lama-cleaner/assets/19190221/f6112fef-477f-42ee-901a-0b4eb2b729da)
![image](https://github.com/Sanster/lama-cleaner/assets/19190221/75ae156c-29d4-4a00-a5ae-273c3ed5fa54)

**Screenshots**

**System Info**
Software version used
- lama-cleaner: 
- pytorch: 
- CUDA:

**Comment by Sanster** [comment]
`--local-files-only` should used after the model has been successfully downloaded，it takes effect through environment variables, so it does not need to be passed to the `from_pretrained`() function. https://github.com/Sanster/lama-cleaner/blob/316198a97aee5a1429ca7f4ff7ca26314ee089e8/iopaint/cli.py#L154

Could you share the complete error log? I tested it locally and if the model has already been downloaded, adding `--local-files-only` should not cause an error.

**Comment by huangzhike** [comment]
```
┌───────────────────── Traceback (most recent call last) ─────────────────────┐
│ E:\work_space\lama-cleaner\iopaint\cli.py:169 in start                      │
│                                                                             │
│   166 │   from iopaint.schema import ApiConfig                              │
│   167 │                                                                     │
│   168 │   app = FastAPI()                                                   │
│ > 169 │   api = Api(                                                        │
│   170 │   │   app,                                                          │
│   171 │   │   ApiConfig(                                                    │
│   172 │   │   │   host=host,                                                │
│                                                                             │
│ E:\work_space\lama-cleaner\iopaint\api.py:152 in __init__                   │
│                                                                             │
│   149 │   │                                                                 │
│   150 │   │   self.file_manager = self._build_file_manager()                │
│   151 │   │   self.plugins = self._build_plugins()                          │
│ > 152 │   │   self.model_manager = self._build_model_manager()              │
│   153 │   │                                                                 │
│   154 │   │   # fmt: off                                                    │
│   155 │   │   self.add_api_route("/api/v1/gen-info", self.api_geninfo, meth │
│                                                                             │
│ E:\work_space\lama-cleaner\iopaint\api.py:355 in _build_model_manager       │
│                                                                             │
│   352 │   │   )                                                             │
│   353 │                                                                     │
│   354 │   def _build_model_manager(self):                                   │
│ > 355 │   │   return ModelManager(                                          │
│   356 │   │   │   name=self.config.model,                                   │
│   357 │   │   │   device=torch.device(self.config.device),                  │
│   358 │   │   │   no_half=self.config.no_half,                              │
│                                                                             │
│ E:\work_space\lama-cleaner\iopaint\model_manager.py:32 in __init__          │
│                                                                             │
│    29 │   │   ):                                                            │
│    30 │   │   │   controlnet_method = self.available_models[name].controlne │
│    31 │   │   self.controlnet_method = controlnet_method                    │
│ >  32 │   │   self.model = self.init_model(name, device, **kwargs)          │
│    33 │                                                                     │
│    34 │   @property                                                         │
│    35 │   def current_model(self) -> ModelInfo:                             │
│                                                                             │
│ E:\work_space\lama-cleaner\iopaint\model_manager.py:56 in init_model        │
│                                                                             │
│    53 │   │   if model_info.support_controlnet and self.enable_controlnet:  │
│    54 │   │   │   return ControlNet(device, **kwargs)                       │
│    55 │   │   elif model_info.name in models:                               │
│ >  56 │   │   │   return models[name](device, **kwargs)                     │
│    57 │   │   else:                                                         │
│    58 │   │   │   if model_info.model_type in [                             │
│    59 │   │   │   │   ModelType.DIFFUSERS_SD_INPAINT,                       │
│                                                                             │
│ E:\work_space\lama-cleaner\iopaint\model\base.py:279 in __init__            │
│                                                                             │
│   276 │   def __init__(self, device, **kwargs):                             │
│   277 │   │   self.model_info = kwargs["model_info"]                        │
│   278 │   │   self.model_id_or_path = self.model_info.path                  │
│ > 279 │   │   super().__init__(device, **kwargs)                            │
│   280 │                                                                     │
│   281 │   @torch.no_grad()                                                  │
│   282 │   def __call__(self, image, mask, config: InpaintRequest):          │
│                                                                             │
│ E:\work_space\lama-cleaner\iopaint\model\base.py:35 in __init__             │
│                                                                             │
│    32 │   │   """                                                           │
│    33 │   │   device = switch_mps_device(self.name, device)                 │
│    34 │   │   self.device = device                                          │
│ >  35 │   │   self.init_model(device, **kwargs)                             │
│    36 │                                                                     │
│    37 │   @abc.abstractmethod                                               │
│    38 │   def init_model(self, device, **kwargs):                           │
│                                                                             │
│ E:\work_space\lama-cleaner\iopaint\model\sdxl.py:48 in init_model           │
│                                                                             │
│    45 │   │   │   │   │   "madebyollin/sdxl-vae-fp16-fix", torch_dtype=torc │
│    46 │   │   │   │   )                                                     │
│    47 │   │   │   │   model_kwargs["vae"] = vae                             │
│ >  48 │   │   │   self.model = handle_from_pretrained_exceptions(           │
│    49 │   │   │   │   StableDiffusionXLInpaintPipeline.from_pretrained,     │
│    50 │   │   │   │   pretrained_model_name_or_path=self.model_id_or_path,  │
│    51 │   │   │   │   torch_dtype=torch_dtype,                              │
│                                                                             │
│ E:\work_space\lama-cleaner\iopaint\model\utils.py:994 in                    │
│ handle_from_pretrained_exceptions                                           │
│                                                                             │
│    991 │   │   │   │   "If the model has already been downloaded, you can a │
│    992 │   │   │   )                                                        │
│    993 │   │   │   exit(-1)                                                 │
│ >  994 │   │   raise e                                                      │
│    995 │   except Exception as e:                                           │
│    996 │   │   raise e                                                      │
│    997                                                                      │
│                                                                             │
│ E:\work_space\lama-cleaner\iopaint\model\utils.py:976 in                    │
│ handle_from_pretrained_exceptions                                           │
│                                                                             │
│    973                                                                      │
│    974 def handle_from_pretrained_exceptions(func, **kwargs):               │
│    975 │   try:                                                             │
│ >  976 │   │   return func(**kwargs)                                        │
│    977 │   except ValueError as e:                                          │
│    978 │   │   if "You are trying to load the model files of the `variant=f │
│    979 │   │   │   logger.info("variant=fp16 not found, try revision=fp16") │
│                                                                             │
│ E:\work_space\lama-cleaner\venv\lib\site-packages\huggingface_hub\utils\_va │
│ lidators.py:118 in _inner_fn                                                │
│                                                                             │
│   115 │   │   if check_use_auth_token:                                      │
│   116 │   │   │   kwargs = smoothly_deprecate_use_auth_token(fn_name=fn.__n │
│   117 │   │                                                                 │
│ > 118 │   │   return fn(*args, **kwargs)                                    │
│   119 │                                                                     │
│   120 │   return _inner_fn  # type: ignore                                  │
│   121                                                                       │
│                                                                             │
│ E:\work_space\lama-cleaner\venv\lib\site-packages\diffusers\pipelines\pipel │
│ ine_utils.py:1096 in from_pretrained                                        │
│                                                                             │
│   1093 │   │   │   │   │   f'The provided pretrained_model_name_or_path "{p │
│   1094 │   │   │   │   │   " is neither a valid local path nor a valid repo │
│   1095 │   │   │   │   )                                                    │
│ > 1096 │   │   │   cached_folder = cls.download(                            │
│   1097 │   │   │   │   pretrained_model_name_or_path,                       │
│   1098 │   │   │   │   cache_dir=cache_dir,                                 │
│   1099 │   │   │   │   resume_download=resume_download,                     │
│                                                                             │
│ E:\work_space\lama-cleaner\venv\lib\site-packages\huggingface_hub\utils\_va │
│ lidators.py:118 in _inner_fn                                                │
│                                                                             │
│   115 │   │   if check_use_auth_token:                                      │
│   116 │   │   │   kwargs = smoothly_deprecate_use_auth_token(fn_name=fn.__n │
│   117 │   │                                                                 │
│ > 118 │   │   return fn(*args, **kwargs)                                    │
│   119 │                                                                     │
│   120 │   return _inner_fn  # type: ignore                                  │
│   121                                                                       │
│                                                                             │
│ E:\work_space\lama-cleaner\venv\lib\site-packages\diffusers\pipelines\pipel │
│ ine_utils.py:1656 in download                                               │
│                                                                             │
│   1653 │   │   model_info_call_error: Optional[Exception] = None            │
│   1654 │   │   if not local_files_only:                                     │
│   1655 │   │   │   try:                                                     │
│ > 1656 │   │   │   │   info = model_info(pretrained_model_name, token=token │
│   1657 │   │   │   except HTTPError as e:                                   │
│   1658 │   │   │   │   logger.warn(f"Couldn't connect to the Hub: {e}.\nWil │
│   1659 │   │   │   │   local_files_only = True                              │
│                                                                             │
│ E:\work_space\lama-cleaner\venv\lib\site-packages\huggingface_hub\utils\_va │
│ lidators.py:118 in _inner_fn                                                │
│                                                                             │
│   115 │   │   if check_use_auth_token:                                      │
│   116 │   │   │   kwargs = smoothly_deprecate_use_auth_token(fn_name=fn.__n │
│   117 │   │                                                                 │
│ > 118 │   │   return fn(*args, **kwargs)                                    │
│   119 │                                                                     │
│   120 │   return _inner_fn  # type: ignore                                  │
│   121                                                                       │
│                                                                             │
│ E:\work_space\lama-cleaner\venv\lib\site-packages\huggingface_hub\hf_api.py │
│ :2084 in model_info                                                         │
│                                                                             │
│   2081 │   │   │   params["securityStatus"] = True                          │
│   2082 │   │   if files_metadata:                                           │
│   2083 │   │   │   params["blobs"] = True                                   │
│ > 2084 │   │   r = get_session().get(path, headers=headers, timeout=timeout │
│   2085 │   │   hf_raise_for_status(r)                                       │
│   2086 │   │   data = r.json()                                              │
│   2087 │   │   return ModelInfo(**data)                                     │
│                                                                             │
│ C:\Users\RTX4090\AppData\Local\Programs\Python\Python310\lib\site-packages\ │
│ requests\sessions.py:600 in get                                             │
│                                                                             │
│   597 │   │   """                                                           │
│   598 │   │                                                                 │
│   599 │   │   kwargs.setdefault("allow_redirects", True)                    │
│ > 600 │   │   return self.request("GET", url, **kwargs)                     │
│   601 │                                                                     │
│   602 │   def options(self, url, **kwargs):                                 │
│   603 │   │   r"""Sends a OPTIONS request. Returns :class:`Response` object │
│                                                                             │
│ C:\Users\RTX4090\AppData\Local\Programs\Python\Python310\lib\site-packages\ │
│ requests\sessions.py:587 in request                                         │
│                                                                             │
│   584 │   │   │   "allow_redirects": allow_redirects,                       │
│   585 │   │   }                                                             │
│   586 │   │   send_kwargs.update(settings)                                  │
│ > 587 │   │   resp = self.send(prep, **send_kwargs)                         │
│   588 │   │                                                                 │
│   589 │   │   return resp                                                   │
│   590                                                                       │
│                                                                             │
│ C:\Users\RTX4090\AppData\Local\Programs\Python\Python310\lib\site-packages\ │
│ requests\sessions.py:701 in send                                            │
│                                                                             │
│   698 │   │   start = preferred_clock()                                     │
│   699 │   │                                                                 │
│   700 │   │   # Send the request                                            │
│ > 701 │   │   r = adapter.send(request, **kwargs)                           │
│   702 │   │                                                                 │
│   703 │   │   # Total elapsed time of the request (approximately)           │
│   704 │   │   elapsed = preferred_clock() - start                           │
│                                                                             │
│ E:\work_space\lama-cleaner\venv\lib\site-packages\huggingface_hub\utils\_ht │
│ tp.py:78 in send                                                            │
│                                                                             │
│    75                                                                       │
│    76 class OfflineAdapter(HTTPAdapter):                                    │
│    77 │   def send(self, request: PreparedRequest, *args, **kwargs) -> Resp │
│ >  78 │   │   raise OfflineModeIsEnabled(                                   │
│    79 │   │   │   f"Cannot reach {request.url}: offline mode is enabled. To │
│    80 │   │   )                                                             │
│    81                                                                       │
└─────────────────────────────────────────────────────────────────────────────┘
OfflineModeIsEnabled: Cannot reach 
https://huggingface.co/api/models/diffusers/stable-diffusion-xl-1.0-inpainting-
0.1: offline mode is enabled. To disable it, please unset the `HF_HUB_OFFLINE` 
environment variable.
```

**Comment by huangzhike** [comment]
pip list
```
Package                Version
---------------------- ------------------------
accelerate             0.26.1
annotated-types        0.6.0
antlr4-python3-runtime 4.9.3
anyio                  4.2.0
bidict                 0.22.1
certifi                2022.12.7
charset-normalizer     2.1.1
click                  8.1.7
colorama               0.4.6
controlnet-aux         0.0.3
diffusers              0.25.0
einops                 0.7.0
exceptiongroup         1.2.0
fastapi                0.108.0
filelock               3.9.0
fsspec                 2023.12.2
h11                    0.14.0
huggingface-hub        0.20.2
idna                   3.4
imageio                2.33.1
importlib-metadata     7.0.1
Jinja2                 3.1.2
lazy_loader            0.3
loguru                 0.7.2
markdown-it-py         3.0.0
MarkupSafe             2.1.3
mdurl                  0.1.2
mpmath                 1.3.0
networkx               3.0
ninja                  1.11.1.1
numpy                  1.24.1
omegaconf              2.3.0
opencv-python          4.9.0.80
packaging              23.2
peft                   0.7.1
piexif                 1.1.3
Pillow                 9.3.0
pip                    23.2.1
psutil                 5.9.7
pydantic               2.5.3
pydantic_core          2.14.6
Pygments               2.17.2
python-engineio        4.8.2
python-multipart       0.0.6
python-socketio        5.7.2
PyYAML                 6.0.1
regex                  2023.12.25
requests               2.28.1
rich                   13.7.0
safetensors            0.4.1
scikit-image           0.22.0
scipy                  1.11.4
setuptools             68.2.0
simple-websocket       1.0.0
sniffio                1.3.0
starlette              0.32.0.post1
sympy                  1.12
tifffile               2023.12.9
timm                   0.9.12
tokenizers             0.15.0
torch                  2.1.1+cu118
torchaudio             2.1.1+cu118
torchvision            0.16.1+cu118
tqdm                   4.66.1
transformers           4.36.2
typer                  0.9.0
typing_extensions      4.9.0
urllib3                1.26.13
uvicorn                0.25.0
wheel                  0.41.2
win32-setctime         1.1.0
wsproto                1.2.0
xformers               0.0.24+6600003.d20240105
yacs                   0.1.8
zipp                   3.17.0
```

**Comment by huangzhike** [comment]
i think its for network reason, even with `local_files_only`, it will still request huggingface, its a problem in China

**Comment by huangzhike** [comment]
also when using stable-diffusion-xl-1.0-inpainting-0.1 with Extender &  LCM Lora, with `local_files_only`, an error occurred:

```
Traceback (most recent call last):
  File "E:\work_space\lama-cleaner\venv\lib\site-packages\anyio\streams\memory.py", line 97, in receive
    return self.receive_nowait()
  File "E:\work_space\lama-cleaner\venv\lib\site-packages\anyio\streams\memory.py", line 92, in receive_nowait
    raise WouldBlock
anyio.WouldBlock

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "E:\work_space\lama-cleaner\venv\lib\site-packages\starlette\middleware\base.py", line 159, in call_next
    message = await recv_stream.receive()
  File "E:\work_space\lama-cleaner\venv\lib\site-packages\anyio\streams\memory.py", line 112, in receive
    raise EndOfStream
anyio.EndOfStream

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "E:\work_space\lama-cleaner\iopaint\api.py", line 108, in exception_handling
    return await call_next(request)
  File "E:\work_space\lama-cleaner\venv\lib\site-packages\starlette\middleware\base.py", line 165, in call_next
    raise app_exc
  File "E:\work_space\lama-cleaner\venv\lib\site-packages\starlette\middleware\base.py", line 151, in coro
    await self.app(scope, receive_or_disconnect, send_no_error)
  File "E:\work_space\lama-cleaner\venv\lib\site-packages\starlette\middleware\exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "E:\work_space\lama-cleaner\venv\lib\site-packages\starlette\_exception_handler.py", line 55, in wrapped_app
    raise exc
  File "E:\work_space\lama-cleaner\venv\lib\site-packages\starlette\_exception_handler.py", line 44, in wrapped_app
    await app(scope, receive, sender)
  File "E:\work_space\lama-cleaner\venv\lib\site-packages\starlette\routing.py", line 746, in __call__
    await route.handle(scope, receive, send)
  File "E:\work_space\lama-cleaner\venv\lib\site-packages\starlette\routing.py", line 288, in handle
    await self.app(scope, receive, send)
  File "E:\work_space\lama-cleaner\venv\lib\site-packages\starlette\routing.py", line 75, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  File "E:\work_space\lama-cleaner\venv\lib\site-packages\starlette\_exception_handler.py", line 55, in wrapped_app
    raise exc
  File "E:\work_space\lama-cleaner\venv\lib\site-packages\starlette\_exception_handler.py", line 44, in wrapped_app
    await app(scope, receive, sender)
  File "E:\work_space\lama-cleaner\venv\lib\site-packages\starlette\routing.py", line 70, in app
    response = await func(request)
  File "E:\work_space\lama-cleaner\venv\lib\site-packages\fastapi\routing.py", line 299, in app
    raise e
  File "E:\work_space\lama-cleaner\venv\lib\site-packages\fastapi\routing.py", line 294, in app
    raw_response = await run_endpoint_function(
  File "E:\work_space\lama-cleaner\venv\lib\site-packages\fastapi\routing.py", line 193, in run_endpoint_function
    return await run_in_threadpool(dependant.call, **values)
  File "E:\work_space\lama-cleaner\venv\lib\site-packages\starlette\concurrency.py", line 35, in run_in_threadpool
    return await anyio.to_thread.run_sync(func, *args)
  File "E:\work_space\lama-cleaner\venv\lib\site-packages\anyio\to_thread.py", line 56, in run_sync
    return await get_async_backend().run_sync_in_worker_thread(
  File "E:\work_space\lama-cleaner\venv\lib\site-packages\anyio\_backends\_asyncio.py", line 2134, in run_sync_in_worker_thread
    return await future
  File "E:\work_space\lama-cleaner\venv\lib\site-packages\anyio\_backends\_asyncio.py", line 851, in run
    result = context.run(func, *args)
  File "E:\work_space\lama-cleaner\iopaint\api.py", line 239, in api_inpaint
    rgb_np_img = self.model_manager(image, mask, req)
  File "C:\Users\RTX4090\AppData\Local\Programs\Python\Python310\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "E:\work_space\lama-cleaner\iopaint\model_manager.py", line 86, in __call__
    self.enable_disable_lcm_lora(config)
  File "E:\work_space\lama-cleaner\iopaint\model_manager.py", line 185, in enable_disable_lcm_lora
    self.model.model.load_lora_weights(self.model.lcm_lora_id)
  File "E:\work_space\lama-cleaner\venv\lib\site-packages\diffusers\loaders\lora.py", line 1441, in load_lora_weights
    state_dict, network_alphas = self.lora_state_dict(
  File "E:\work_space\lama-cleaner\venv\lib\site-packages\huggingface_hub\utils\_validators.py", line 118, in _inner_fn
    return fn(*args, **kwargs)
  File "E:\work_space\lama-cleaner\venv\lib\site-packages\diffusers\loaders\lora.py", line 235, in lora_state_dict
    weight_name = cls._best_guess_weight_name(
  File "E:\work_space\lama-cleaner\venv\lib\site-packages\diffusers\loaders\lora.py", line 307, in _best_guess_weight_name
    raise ValueError("When using the offline mode, you must specify a `weight_name`.")
ValueError: When using the offline mode, you must specify a `weight_name`.
```

**Comment by Sanster** [comment]
> i think it's for network reason, even with `local_files_only`, it will still request huggingface, which it's a problem in China

感谢反馈问题，我能够复现这个问题，不是网络问题，发现是新版 diffusers 的问题， https://github.com/huggingface/diffusers/pull/1767#issuecomment-1893417458，后续我会把 `local_files_only` 给 `from_pretrained` 加上来修复这个问题

**Comment by Sanster** [comment]
fixed in `1.0.0b8`
